### PR TITLE
Move filter handling from modify to applyFilter

### PR DESCRIPTION
**Pull request by lehni**
As discussed on [Gitter](https://gitter.im/Vincit/objection.js?at=59f01c8ad6c36fca3192f499), here now the PR with documentation

### CODE REVIEW

File: lib/queryBuilder/QueryBuilder.js [hunk 1]
```diff
@@ -130,19 +130,19 @@ class QueryBuilder extends QueryBuilderBase {
...
+      const filter = namedFilters[arguments[i]];
+      filter(this);
     }
+    return this;
+  }
+
+  modify() {
+    return typeof arguments[0] === 'string'
+      ? this.applyFilter.apply(this, arguments)
+      : super.modify.apply(this, arguments);
```
Comment (lehni) [inline lib/queryBuilder/QueryBuilder.js]: I don't know if you dig ternary operators or not... I do, as you can see here : )
Happy to rewrite it with if / else if you prefer.

**Comment by koskimas** [comment]
Looks great. This needs to have at least one test somewhere. Then we're good to go.

File: lib/queryBuilder/QueryBuilder.js (hunk 1 above)
Comment (koskimas) [reply lib/queryBuilder/QueryBuilder.js]: I also like ternary operators when they make stuff simpler. I wish ecmascript handled `if`, `try` etc. blocks as expressions that return a value like kotlin does:
```js
const foo = if (x < 10) "bar" else "baz"
const bar = try {
  doStuff();
} catch (err) {
  null
}
```

Comment (lehni) [reply lib/queryBuilder/QueryBuilder.js]: Yeah that would be nice. CoffeeScript does this also, and I believe Ruby and a bunch of others too.

**Comment by lehni** [comment]
@koskimas done: 95a5d0cd45c8cafcf053441e8abfd57a70dfa860

**Comment by koskimas** [comment]
Looks good, thanks!

**Comment by lehni** [comment]
@koskimas I've been thinking, wouldn't `mergeFilter()` have been the better name for this? 

There is already `mergeEager()`, and the effect of applying a filter is similar: Previous filters don't get erased, they get "merged"...

**Pull request merged**